resolver/src/main/java/io/netty/resolver/HostsFileParser.java

```diff
@@ -0,0 +1,158 @@
+ * A parser for hosts files.
+ */
+public final class HostsFileParser {
+
+    private static final String WINDOWS_HOSTS_FILE_PATH = "C:\\Windows\\system32\\drivers\\etc\\hosts";
+    private static final String X_PLATFORMS_HOSTS_FILE_PATH = "/etc/hosts";
+
+    private static final InternalLogger logger = InternalLoggerFactory.getInstance(HostsFileParser.class);
+
+    /**
+     * Parse hosts file at standard OS location.
+     *
+     * @return a map of hostname or alias to {@link InetAddress}
+     */
+    public static Map<String, InetAddress> parseSilently() {
+        String path = PlatformDependent.isWindows() ? WINDOWS_HOSTS_FILE_PATH : X_PLATFORMS_HOSTS_FILE_PATH;
+        try {
+            return parse(new File(path));
+        } catch (IOException e) {
+            logger.error("Failed to load and parse hosts file", e);
+            return Collections.emptyMap();
+        }
+    }
+
+    /**
+     * Parse hosts file at standard OS location.
+     *
+     * @return a map of hostname or alias to {@link InetAddress}
+     * @throws IOException file could not be read
+     */
+    public static Map<String, InetAddress> parse() throws IOException {
+        String path = PlatformDependent.isWindows() ? WINDOWS_HOSTS_FILE_PATH : X_PLATFORMS_HOSTS_FILE_PATH;
+        return parse(new File(path));
+    }
+
+    /**
+     * Parse a hosts file.
+     *
+     * @param file the file to be parsed
+     * @return a map of hostname or alias to {@link InetAddress}
+     * @throws IOException file could not be read
+     */
+    public static Map<String, InetAddress> parse(File file) throws IOException {
+        checkNotNull(file, "file");
+        if (file.exists() && file.isFile()) {
+            return parse(new BufferedReader(new FileReader(file)));
+        } else {
+            return Collections.emptyMap();
+        }
+    }
+
+    /**
+     * Parse a reader of hosts file format.
+     *
+     * @param reader the file to be parsed
+     * @return a map of hostname or alias to {@link InetAddress}
+     * @throws IOException file could not be read
+     */
+    public static Map<String, InetAddress> parse(Reader reader) throws IOException {
+        checkNotNull(reader, "reader");
+        BufferedReader buff = new BufferedReader(reader);
+        try {
+            Map<String, InetAddress> entries = new HashMap<String, InetAddress>();
+            String line;
+            while ((line = buff.readLine()) != null) {
+                // remove comment
+                int commentPosition = line.indexOf('#');
+                if (commentPosition != -1) {
+                    line = line.substring(0, commentPosition);
+                }
+                // skip empty lines
+                line = line.trim();
+                if (line.isEmpty()) {
+                    continue;
+                }
+
+                // split
+                List<String> lineParts = new ArrayList<String>();
+                for (String s: line.split("[ \t]+")) {
```

Comment:
StringUtil.split only supports a single char delimiter. Hosts files can use both spaces and tabs (note the space in the pattern).